staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_work_estimator.go

```diff
@@ -57,7 +57,14 @@ func (e *listWorkEstimator) estimate(r *http.Request, flowSchemaName, priorityLe
 		// return maximumSeats for this request to be consistent.
 		return WorkEstimate{InitialSeats: maximumSeats}
 	}
-	isListFromCache := !shouldListFromStorage(query, &listOptions)
+	isListFromCache := !storage.ShouldDelegateList(storage.ListOptions{
```

Comment:
This is potentially risky change.
If at some point we would decide to use other fields of ListOptions in the function (like selectors), those aren't set here. Also - we didn't have to create storage.ListOptions here, which adds some cost.

So I'm not 100% convinced we want this change.

Reply:
@wojtek-t 

> If at some point we would decide to use other fields of ListOptions in the function (like selectors), those aren't set here

That's a fair point. How about having a conversion function from `metav1.ListOptions` -> `storage.ListOptions` that is used in these places?

> we didn't have to create storage.ListOptions here, which adds some cost

As far as I understand, we wouldn't get any additional allocs. Everything would happen through the stack. Would a benchmark help?

Trying to think of ways to move forward because I do think that keeping all sources in sync with the logic in that function would get difficult, I am currently aware of 2 places (flow control and integration test), its possible that other places will come up.

Reply:
Conversion func would help, although I would like it to be constructed in a way that it could also be used in:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L356

> Would a benchmark help?

Yeah - some small benchmark would be useful here.